### PR TITLE
feat: SNI Support

### PR DESCRIPTION
**Pull request by DownD**
Closes #

## Proposed Changes

Added an option to change the SNI field in the TLS header and the Host field in the HTTP header.
It can be used on servers behind NGINX using the SNI field for redirection.

## Checklist

- [X] CHANGELOG.md updated
- [X] Rebased/mergeable
- [x] `pytest tests` completes successfully
- [ ] Commit messages are in [semantic format](https://seesparkbox.com/foundry/semantic_commit_messages)
- [ ] Sign [CLA](https://www.influxdata.com/legal/cla/) (if not already signed)

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -193,6 +193,7 @@ The following options are supported:
 - ``connection_pool_maxsize`` - set the number of connections to save that can be reused by urllib3
 - ``auth_basic`` - enable http basic authentication when talking to a InfluxDB 1.8.x without authentication but is accessed via reverse proxy with basic authentication (defaults to false)
 - ``profilers`` - set the list of enabled `Flux profilers <https://docs.influxdata.com/influxdb/v2.0/reference/flux/stdlib/profiler/>`_
+- ``server_hostname`` - custom server hostname to be passed on the HTTP Host request and on the TLS Server Name Identification
```
Comment (bednar) [inline README.rst:196]: Please move `server_hostname` over to `profilers`.

File: influxdb_client/client/influxdb_client.py [hunk 2]
```diff
@@ -53,6 +53,7 @@ def __init__(self, url, token, debug=None, timeout=10_000, enable_gzip=False, or
                               does not use auth-enabled but is protected by a reverse proxy with basic authentication.
                               (defaults to false, don't set to true when talking to InfluxDB 2)
         :key list[str] profilers: list of enabled Flux profilers
+        :key str server_hostname: Sets the SNI field, if using TLS, and sets the Host header to a custom host.
```
Comment (bednar) [inline influxdb_client/client/influxdb_client.py:56]: Please move `server_hostname` over to `profilers`.

File: influxdb_client/client/influxdb_client.py [hunk 3]
```diff
@@ -207,10 +210,14 @@ def config_value(key: str):
         if config.has_option('influx2', 'proxy'):
             proxy = config_value('proxy')
 
+        server_hostname = None
+        if config.has_option('influx2', 'server_hostname'):
+            server_hostname = config_value(server_hostname)
```
Comment (bednar) [inline influxdb_client/client/influxdb_client.py:215]:
```suggestion
            server_hostname = config_value('server_hostname')
```

**Review by bednar — CHANGES_REQUESTED** [review]
Thanks for your PR 👍 
There are a few requirements that must be be satisfy before we accept the PR:

1. Update the `CHANGELOG.md` 
2. Please add tests for your changes

and also:

File: README.rst (hunk 1 above)
Comment (DownD) [reply README.rst:196]: Sorry my ignorance has i never used the profilers field, but from a fast a search i can see that this is related to profilling tools which is not related to this request.
Could you plesae provide additional insights on your final goal of this change?

**Comment by DownD** [comment]
I would like to ask your opinion on the test cases that i submitted now.
Currently this change is being used to redirect the requests to a specific server using an NGINX proxy server based on Server Name Indication (SNI) field of the TLS layer.
I don't see a way, using only python, to check the SNI field before sending the request in order to build a unit test.

File: README.rst (hunk 1 above)
Comment (bednar) [reply README.rst:196]: It's just a grouping of settings with the same meaning - `connection_pool_maxsize `, `auth_basic`, `server_hostname` - are related to HTTP.

File: influxdb_client/client/influxdb_client.py [hunk 4]
```diff
@@ -207,10 +210,14 @@ def config_value(key: str):
         if config.has_option('influx2', 'proxy'):
             proxy = config_value('proxy')
 
+        server_hostname = None
```
Comment (bednar) [inline influxdb_client/client/influxdb_client.py:213]: Please add `server_hostname` into doc string:

image

File: influxdb_client/client/influxdb_client.py [hunk 5]
```diff
@@ -207,10 +210,14 @@ def config_value(key: str):
...
+            server_hostname = config_value('server_hostname')
+
         return cls(url, token, debug=debug, timeout=_to_int(timeout), org=org, default_tags=default_tags,
                    enable_gzip=enable_gzip, verify_ssl=_to_bool(verify_ssl), ssl_ca_cert=ssl_ca_cert,
                    connection_pool_maxsize=_to_int(connection_pool_maxsize), auth_basic=_to_bool(auth_basic),
-                   profilers=profilers, proxy=proxy)
+                   profilers=profilers, proxy=proxy, server_hostname=server_hostname)
 
     @classmethod
     def from_env_properties(cls, debug=None, enable_gzip=False):
```
Comment (bednar) [inline influxdb_client/client/influxdb_client.py:223]: Please add possibility to configure `server_hostname` from env properties.

**Review by bednar — CHANGES_REQUESTED** [review]
> I would like to ask your opinion on the test cases that i submitted now.
Currently this change is being used to redirect the requests to a specific server using an NGINX proxy server based on Server Name Indication (SNI) field of the TLS layer.
I don't see a way, using only python, to check the SNI field before sending the request in order to build a unit test.

The tests you added are pretty enough. Please fix the following requirements and we will be ready to merge your PR:

**Review by bednar — CHANGES_REQUESTED** [review]
@DownD thanks again for your PR. We are almost done. There left only two things before we accept this PR:

1. Please finish our Checklist:
image
2. Rebase your sources with `master` branch and update `CHANGELOG.md` to proper version

**Comment by bednar** [comment]
This PR has been closed because it has not had recent activity. Please reopen if this PR is still important to you and you want to continue with them.

**Pull request closed without merging**